File: EVR/src/main/java/com/group7/evr/service/UserService.java
package com.group7.evr.service;

import com.group7.evr.dto.UserRegistrationRequest;
import com.group7.evr.entity.User;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;
import java.util.Map;

public interface UserService {
    User register(UserRegistrationRequest request,
                         MultipartFile personalIdImage,
                  MultipartFile licenseImage) throws IOException;

    User verifyEmailToken(String token);

    User verifyUser(Integer userId, User staff);

    User getUserById(Integer userId);

    Map<String, Object> login(String email, String password);

    User updateUser(Integer userId, User userUpdates);

    void changePassword(Integer userId, String currentPassword, String newPassword);

    Map<String, Object> getAllUsers(int page, int size, String role, String status);

    List<User> getRiskUsers();

    User updateUserStatus(Integer userId, String status, String reason);

    User createStaff(User staff);

    List<User> getStaff(Integer stationId);

    User getPrimaryStaffForStation(Integer stationId);

    User updateStaff(Integer staffId, User staffUpdates);

    Map<String, Object> deleteStaff(Integer staffId);

    void logAudit(User user, String action);
}

File: EVR-FE/src/services/user.service.js
import apiClient from './api/axios';
import { ENDPOINTS } from './api/endpoints';

export const userService = {
  getUserById: async (id) => {
    return apiClient.get(ENDPOINTS.GET_USER(id));
  },

  updateUser: async (id, userData) => {
    return apiClient.put(ENDPOINTS.UPDATE_USER(id), userData);
  },

  changePassword: async (id, payload) => {
    return apiClient.put(ENDPOINTS.CHANGE_PASSWORD(id), payload);
  },

  verifyUser: async (id, staff) => {
    return apiClient.put(ENDPOINTS.VERIFY_USER(id), staff);
  },

  getAllUsers: async (page = 0, size = 10, role = null, status = null) => {
    const params = { page, size };
    if (role) params.role = role;
    if (status) params.status = status;
    return apiClient.get(ENDPOINTS.GET_ALL_USERS, { params });
  },

  getRiskUsers: async () => {
    return apiClient.get(ENDPOINTS.GET_RISK_USERS);
  },

  updateUserStatus: async (id, status, reason = null) => {
    const params = { status };
    if (reason) params.reason = reason;
    return apiClient.put(ENDPOINTS.UPDATE_USER_STATUS(id), null, { params });
  },
};

